File: vitalcore-expo/e2e/pages/OnboardingPage.ts
import BasePage from './BasePage';

export default class OnboardingPage extends BasePage {
  public get fullNameInput() { return this.getByTestId('onboarding-fullname-input'); }
  public get ageInput() { return this.getByTestId('onboarding-age-input'); }
  public get genderMaleBtn() { return this.getByTestId('onboarding-gender-male'); }
  public get genderFemaleBtn() { return this.getByTestId('onboarding-gender-female'); }
  public get heightInput() { return this.getByTestId('onboarding-height-input'); }
  public get weightInput() { return this.getByTestId('onboarding-weight-input'); }
  public get nextBtn() { return this.getByTestId('onboarding-next-btn'); }
  public get backBtn() { return this.getByTestId('onboarding-back-btn'); }
  public get completeBtn() { return this.getByTestId('onboarding-complete-btn'); }

  public async fillStep1(fullName: string, age: string): Promise<void> {
    await this.waitAndSetValue('onboarding-fullname-input', fullName);
    await this.waitAndSetValue('onboarding-age-input', age);
    await this.waitAndClick('onboarding-gender-male');
    await this.waitAndClick('onboarding-next-btn');
  }

  public async fillStep2(height: string, weight: string): Promise<void> {
    await this.waitAndSetValue('onboarding-height-input', height);
    await this.waitAndSetValue('onboarding-weight-input', weight);
    await this.waitAndClick('onboarding-next-btn');
  }
}

File: test_pages/onboarding_page.py
"""Page Object Model for Health Onboarding (/auth/onboarding)."""

from selenium.webdriver.common.by import By
from test_pages.base_page import BasePage

class OnboardingPage(BasePage):
    NEXT_BUTTON = (By.XPATH, "//button[contains(text(), 'Next') or contains(text(), 'Continue')]")
    BACK_BUTTON = (By.XPATH, "//button[contains(text(), 'Back')]")
    COMPLETE_BUTTON = (By.XPATH, "//button[contains(text(), 'Complete') or contains(text(), 'Finish')]")
    STEP_HEADER = (By.XPATH, "//h1 | //h2 | //*[contains(text(), 'Step')]")
    
    # Inputs
    AGE_INPUT = (By.NAME, "age")
    HEIGHT_INPUT = (By.NAME, "height")
    WEIGHT_INPUT = (By.NAME, "weight")
    GENDER_MALE = (By.XPATH, "//button[contains(text(), 'Male') or @value='male']")
    GENDER_FEMALE = (By.XPATH, "//button[contains(text(), 'Female') or @value='female']")
    MEDICAL_SEARCH_INPUT = (By.XPATH, "//input[@placeholder='Search conditions...']")

    def open(self):
        self.navigate_to("/auth/onboarding")

    def click_next(self):
        self.click(*self.NEXT_BUTTON)

    def click_back(self):
        self.click(*self.BACK_BUTTON)

    def click_complete(self):
        self.click(*self.COMPLETE_BUTTON)
